docs/management_api.yml

```diff
@@ -28,83 +28,80 @@ tags:
   - name: ManagementAPI
 
 paths:
-  /configuration:
+  /config/devices/{deviceId}:
     get:
-      operationId: List Parameters
+      operationId: Get key-value pair store
       tags:
         - ManagementAPI
-      summary: Get existing parameters
-      # parameters: {}
+      summary: Query the configuration store; retrieve all key-value pairs
+      parameters:
+        - in: path
+          name: deviceId
+          schema:
+            type: string
+          required: true
+          description: ID of the device to query.
       responses:
         200:
           description: OK
           content:
             application/json:
               schema:
-                type: array
-                items:
-                  $ref: '#/components/schemas/Parameter'
+                type: object
+                properties:
+                  id:
+                    type: string
+                  expected:
+                    type: array
+                    items:
+                      $ref: '#/components/schemas/Parameter'
+                  actual:
+                    type: array
```

Comment:
I see you implemented the set/get interface using an array. I would suggest using a free object here, which makes it possible in the future to evolve to full JSON if needed. Instead if sending `[{"key": "a", "value": "b"}]` I would like to be able to send `{"a": "b"}`. This applies both to the GET and to the PUT method. I know this changes your implementation a lot, but it is way more user-friendly and flexible than a list of key/value attributes. By the way, in the current implementation you can provide the same key multiple times providing multiple items with the same key, the latest is the one inserted.

Reply:
I know that, and my original agreement with Manuel was to use the exactly what you say. however, what I did was dictated by what we already have in the code:
```
DesiredAttributes []Attribute
```
ok, I will return to the original idea.

Reply:
back to the first implementation. in other words: fixed.